### PR TITLE
LF-3190-change-too-many-password-resets-to-be-in-line-error-rather-than-snack-bar

### PR DESCRIPTION
**Pull request by navDhammu**
**Description**

Replaced the error snackbar with an inline error message by creating a new state variable in redux to hold the error message. 

Modified modal component to accept an error prop to be rendered underneath the button.

Jira link: https://lite-farm.atlassian.net/browse/LF-3190

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/src/components/Modals/ModalComponent/v1/index.jsx [hunk 1]
```diff
@@ -26,7 +27,9 @@ export default function ModalComponent({
         {title}
       </Title>
       {descriptions.map((description) => (
-        <Title style={{ marginBottom: 0, textAlign: 'center' }}>{description}</Title>
+        <Title key={description} style={{ marginBottom: 0, textAlign: 'center' }}>
```
Comment (navDhammu) [inline packages/webapp/src/components/Modals/ModalComponent/v1/index.jsx:30]: added the key here because pre-commit / linting was giving an error

**Review by Duncan-Brain — COMMENTED** [review]
Looks good to me! I am just wondering though if it should dismiss the error on click so that the user knows it is retrying?

**Comment by navDhammu** [comment]
> Looks good to me! I am just wondering though if it should dismiss the error on click so that the user knows it is retrying?

Would it be better to just disable the resend button when theres an error? Because clicking the button would result in the same error again and again.

**Comment by kathyavini** [comment]
@navDhammu and @Duncan-Brain personally I like the idea of disabling the button!

**Comment by Duncan-Brain** [comment]
I like the idea too -- could be worth asking/tagging Kevin and Loic on Jira ticket?

File: packages/webapp/src/components/Modals/ModalComponent/v1/index.jsx [hunk 2]
```diff
@@ -37,6 +40,7 @@ export default function ModalComponent({
       >
         {buttonLabel}
       </Button>
+      {error && <span className={styles.error}>{error}</span>}
```
Comment (antsgar) [inline packages/webapp/src/components/Modals/ModalComponent/v1/index.jsx:43]: I'd suggest using the `Error` component from the `Typography` file here which will already give you the right styling (the font color is a deeper red than the regular red used here), and then using custom styling only for the margin.

**Review by antsgar — CHANGES_REQUESTED** [review]

File: packages/webapp/src/components/Modals/ModalComponent/v1/index.jsx (hunk 2 above)
Comment (navDhammu) [reply packages/webapp/src/components/Modals/ModalComponent/v1/index.jsx]: Thanks for the suggestion - I went ahead and replaced it with the Error component.

**Review by Duncan-Brain — APPROVED** [review]
Looking good to me! Thanks for updating

File: packages/webapp/src/components/Modals/ModalComponent/v1/index.jsx (hunk 2 above)
Comment (antsgar) [reply packages/webapp/src/components/Modals/ModalComponent/v1/index.jsx]: Thank you for updating this and for the contribution! 🙂

**Review by antsgar — APPROVED** [review]

**Pull request merged**